core-bundle/src/Image/Studio/MetaDataFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Studio;
+
+use Contao\Controller;
+use Contao\CoreBundle\Framework\ContaoFramework;
+
+class MetaDataFactory
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    /**
+     * Create a meta data container from a set of values.
+     *
+     * @param bool $canonicalize Whether the values should be constrained to the current `tl_files.meta` configuration.
+     */
+    public function create(array $values, bool $canonicalize = true): MetaData
+    {
+        if ($canonicalize) {
+            $values = $this->canonicalizeValues($values, $this->getMetaFields());
+        }
+
+        return new MetaData($values);
+    }
+
+    /**
+     * Create an empty meta data container that will contain the default fields
+     * with empty values.
+     */
+    public function createEmpty(): MetaData
+    {
+        return $this->create([]);
+    }
+
+    /**
+     * Get the meta fields registered in `tl_files.meta`.
+     */
+    public function getMetaFields(): array
+    {
+        $this->framework->initialize();
+
+        /** @var Controller $controller */
+        $controller = $this->framework->getAdapter(Controller::class);
+        $controller->loadDataContainer('tl_files');
```

Comment:
Isn’t it problematic to load the tl_files DCA here?  
This would probably occur in every frontend request.

Reply:
:thinking: Can we get to the information without loading the DCA? (we could maybe introduce a config value, that's set in the container)

Reply:
Is it necessary to have these keys? We use them now to filter the fields of `tl_content` right?

Reply:
Yes. Or rather to constrain the `MetaData` in general (because the whole mapping will end up the template). I think we could move the logic to `ContentModel->getMetaData()` but would still need to check `tl_files` there.